Hice la funcion calulaPi #YOLO

charis38 opened the pull request.

charis38 committed:
Hice la funcion calulaPi #YOLO

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+int calculaPi(int num)
```
Pi no debe regresar un entero
¿que es num? quitale ese argumento, calcula para 1000

charis38 replied:
Corregido

zaballin replied:
No regresas un float?

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+
+int calculaPi(int num)
+{
+  float pi = 1, divisor = 3, i, flag = 0;
```
si es flotante debería ser 1.0, 3.0 
i debe ser entero, flag tambien

charis38 replied:
Corregido

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+int calculaPi(int num)
+{
+  float pi = 1, divisor = 3, i, flag = 0;
+  for(i = 0; i < num; i++)
```
comenta el numero de iteraciones

charis38 replied:
Corregido

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+      if(flag == 0)
+	{
+	  pi = pi - (1.0/divisor);
+	  printf("%f %f\n", divisor, pi);
```
aqui no imprimo

charis38 replied:
Corregido

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+	  if(flag  == 1)
+	    {
+	      pi = pi + (1.0/divisor);
+	      printf("%f %f\n", divisor, pi);
```
aqui no imprimo

charis38 replied:
Corregido

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+      divisor = divisor + 2;
+    }
+  pi = pi*4;
+  printf("%f\n", pi);
```
aqui no imprimo

charis38 replied:
Corregido

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+    }
+  pi = pi*4;
+  printf("%f\n", pi);
+}
```
falta un return

charis38 replied:
Corregido

charis38 committed:
Ya no recibe argumentos, fije el numero de terminos que utiliza en 10000.

Corregi pi de 1 a 1.0 porque es flotante.

Cambie a enteros las variables que antes eran flotantes y no necesitaban serlo
(divisor, i, flag y num).

Borre los printfs y agregue un comentario.

charis38 committed:
Corregi el Return

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+#include <stdio.h>
```
Comentarios tipo Doxygen

eortizl commented on pi.c:
```diff
@@ -0,0 +1,28 @@
+{
+  float pi = 1.0; 
+  int divisor = 3, i, flag = 0, num = 10000;
+  //Lo hace 10000 veces para mayor exactitud
```
En realidad no
